xcdat/dataset.py

```diff
@@ -526,6 +590,10 @@ def _postprocess_dataset(
                 "This dataset does not have longitude coordinates to reorient."
             )
 
+    if time_encoding is not None:
+        time_dim = get_axis_dim(dataset, "T")
+        dataset[time_dim].encoding = time_encoding
+
```

Comment:
There were to options for where to include the logic for adding `time_encoding` attrs back to the final dataset:

1. In `_postprocess_dataset()`, which is shown here
   - On the other hand, this function is used by both `open_dataset()` and `open_mfdataset()`, but this logic is only needed for `open_mfdataset()`. 
2. Keep this logic stored in `open_mfdataset()`, based on above reasoning.

I chose 1 since it is a "postprocessing" operation, although I am open to 2 since the logic is specific to `open_mfdataset()`.

I will think about this more.